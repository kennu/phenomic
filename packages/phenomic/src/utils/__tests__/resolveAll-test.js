import resolveAll from "../resolveAll"

describe("resolveAll", () => {

  it("should resolve all values in object", () => {
    return resolveAll({ a: Promise.resolve(1), b: Promise.resolve(2) })
      .then(values => {
        expect(values).toMatchSnapshot()
      })
  })

  it("should resolve non promises values", () => {
    return resolveAll({ a: Promise.resolve(1), b: 2 })
      .then(values => {
        expect(values).toMatchSnapshot()
      })
  })

  it("should reject if a promise rejects", () => {
    return resolveAll({ a: Promise.resolve(1), b: Promise.reject(2) })
      .catch(reason => {
        expect(reason).toMatchSnapshot()
      })
  })

})
