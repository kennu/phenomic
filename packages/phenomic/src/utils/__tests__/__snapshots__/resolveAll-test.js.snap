exports[`resolveAll should reject if a promise rejects 1`] = `2`;

exports[`resolveAll should resolve all values in object 1`] = `
Object {
  "a": 1,
  "b": 2,
}
`;

exports[`resolveAll should resolve non promises values 1`] = `
Object {
  "a": 1,
  "b": 2,
}
`;
